scripts/check_copyright_notice.py

```diff
@@ -30,18 +30,25 @@
 
 import itertools
 import re
+import shutil
+import subprocess  # nosec
 import sys
+from datetime import datetime
 from pathlib import Path
+from typing import List, Tuple
 
 
+GIT_PATH = shutil.which("git")
 SUPPORTED_YEARS_FETCHAI = ["2019", "2020", "2021"]
+SUPPORTED_YEARS_VALORY = ["2021", "2022"]
 
 
-HEADER_REGEX_FETCHAI = fr"""(#!/usr/bin/env python3
+HEADER_REGEX_FETCHAI = re.compile(
+    fr"""(#!/usr/bin/env python3
 )?# -\*- coding: utf-8 -\*-
 # ------------------------------------------------------------------------------
 #
-#   (Copyright 2018-({"|".join(SUPPORTED_YEARS_FETCHAI)}) Fetch.AI Limited|Copyright [0-9]{{4}}(-[0-9]{{4}})? [a-zA-Z_]+)
+#   Copyright ((20\d\d)(-20\d\d)?) Fetch.AI Limited
```

Comment:
There's also files with just 1 year (e.g. 2022, or an old file edited in a single year)?

Reply:
I suppose we cover this :D